package/lib/compileFunctions.js

```diff
@@ -48,6 +52,10 @@ module.exports = {
         funcObject.environment // eslint-disable-line comma-dangle
       );
 
+      if (!funcTemplate.properties.serviceAccountEmail) {
+        delete funcTemplate.properties.serviceAccountEmail;
+      }
```

Comment:
Why is that needed?

Reply:
Because GCP cannot receive the variable as empty, (throws error) we need ro remove it from the whole structure.

Reply:
But you are right... the template does not have it at this stage, so, is not needed actually...

Reply:
Sorry, tested thoroughly and this piece of code needs to be in place when null:
 `       
error: instance type (null) does not match any allowed primitive type (allowed: ["string"])
      level: "error"
      schema: {"loadingURI":"#","pointer":"/create/properties/serviceAccountEmail"}
      instance: {"pointer":"/serviceAccountEmail"}
      domain: "validation"
      keyword: "type"
      found: "null"
      expected: ["string"]
`

Reply:
And what would be the case if we'd fallback to `undefined` instead of `null` ?

Reply:
Falled back to undefined also fails:

<details>
<summary>Click here to see Api Error from GCP</summary>

>   Error: Error parsing configuration: could not determine a constructor for the tag tag:yaml.org,2002:js/undefined
>    in 'string', line 52, column 28:
>             serviceAccountEmail: !<tag:yaml.org,2002:js/undefined> ''
>                                  ^
> 
>       at Gaxios._request (D:\CLOUD_SANDBOX\EVALUESERVE\CODE\CloudFunctions-NodeJS\node_modules\gaxios\src\gaxios.ts:109:15)
>       at process._tickCallback (internal/process/next_tick.js:68:7)

</details>

Reply:
Ok, thanks for investigating that